### PR TITLE
#92 Sync filterable side screens.

### PR DESCRIPTION
**Pull request by zuev93**
Examples:
FoodBox - TreeFilterableSideScreen
LiquidFilter - FilterSideScreen

### CODE REVIEW

File: MultiplayerMod/Game/Mechanics/FilterEvents.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+namespace MultiplayerMod.Game.Mechanics;
+
+public static class FilterEvents {
+
+    public static event Action<MultiplayerReference, Tag> SetFilter;
+    public static event Action<MultiplayerReference, Tag> AddTagToFilter;
+    public static event Action<MultiplayerReference, Tag> RemoveTagFromFilter;
+
+    [HarmonyPatch(typeof(Filterable))]
+    private class FilterableEvents {
```
Comment (polycone) [inline MultiplayerMod/Game/Mechanics/FilterEvents.cs:15]: You can use all these without subclasses, just `[HarmonyPatch(typeof(Filterable), nameof(Filterable.OnFilterChanged))]`

File: MultiplayerMod/Multiplayer/Configuration/GameEventBindings.cs [hunk 2]
```diff
@@ -157,6 +158,10 @@ public class GameEventBindings {
         };
 
         SliderEvents.SetSliderValue += eventArgs => client.Send(new SetSliderValue(eventArgs));
+
+        FilterEvents.SetFilter += (reference, tag) => client.Send(new SetFilter(reference, tag));
+        FilterEvents.AddTagToFilter +=  (reference, tag) => client.Send(new AddTagToFilter(reference, tag));
+        FilterEvents.RemoveTagFromFilter +=  (reference, tag) => client.Send(new RemoveTagFromFilter(reference, tag));
```
Comment (polycone) [inline MultiplayerMod/Multiplayer/Configuration/GameEventBindings.cs:164]: A bit of style issues.

**Review by polycone — APPROVED** [review]

File: MultiplayerMod/Multiplayer/Configuration/GameEventBindings.cs (hunk 2 above)
Comment (zuev93) [reply MultiplayerMod/Multiplayer/Configuration/GameEventBindings.cs:164]: Done

File: MultiplayerMod/Game/Mechanics/FilterEvents.cs (hunk 1 above)
Comment (zuev93) [reply MultiplayerMod/Game/Mechanics/FilterEvents.cs:15]: Done

**Pull request merged**